plugins/modules/fusion_info.py

```diff
@@ -81,260 +81,459 @@
     get_fusion,
     fusion_argument_spec,
 )
+from ansible_collections.purestorage.fusion.plugins.module_utils.errors import (
+    install_fusion_exception_hook,
+)
 import time
 
 
-def _convertMicroseconds(micros):
+def _convert_microseconds(micros):
     seconds = (micros / 1000) % 60
     minutes = (micros / (1000 * 60)) % 60
     hours = (micros / (1000 * 60 * 60)) % 24
     return seconds, minutes, hours
 
 
-def generate_default_dict(fusion):
-    default_info = {}
-
-    arrays_api_instance = purefusion.ArraysApi(fusion)
-    az_api_instance = purefusion.AvailabilityZonesApi(fusion)
-    default_api_instance = purefusion.DefaultApi(fusion)
-    hw_api_instance = purefusion.HardwareTypesApi(fusion)
-    host_access_api_instance = purefusion.HostAccessPoliciesApi(fusion)
-    id_api_instance = purefusion.IdentityManagerApi(fusion)
-    nic_api_instance = purefusion.NetworkInterfacesApi(fusion)
-    nig_api_instance = purefusion.NetworkInterfaceGroupsApi(fusion)
-    plgrp_api_instance = purefusion.PlacementGroupsApi(fusion)
-    protpol_api_instance = purefusion.ProtectionPoliciesApi(fusion)
-    regions_api_instance = purefusion.RegionsApi(fusion)
-    role_assign_api_instance = purefusion.RoleAssignmentsApi(fusion)
-    roles_api_instance = purefusion.RolesApi(fusion)
-    snapshot_api_instance = purefusion.SnapshotsApi(fusion)
-    send_api_instance = purefusion.StorageEndpointsApi(fusion)
-    storage_srv_api_instance = purefusion.StorageServicesApi(fusion)
-    storage_class_api_instance = purefusion.StorageClassesApi(fusion)
-    tenant_api_instance = purefusion.TenantsApi(fusion)
-    tenantspace_api_instance = purefusion.TenantSpacesApi(fusion)
-    vol_api_instance = purefusion.VolumesApi(fusion)
-
-    default_info["version"] = default_api_instance.get_version().version
+def _api_permission_denied_handler(name):
+    """Return decorator which catches #403 errors"""
 
-    storage_services = storage_srv_api_instance.list_storage_services()
-    default_info["storage_services"] = len(storage_services.items)
-    sclass = 0
-    for storserv in range(0, len(storage_services.items)):
-        sclass = sclass + len(
-            storage_class_api_instance.list_storage_classes(
-                storage_service_name=storage_services.items[storserv].name
-            ).items
-        )
-    default_info["storage_classes"] = sclass
+    def inner(func):
+        def wrapper(module, fusion, *args, **kwargs):
+            try:
+                return func(module, fusion, *args, **kwargs)
+            except purefusion.rest.ApiException as exc:
+                if exc.status == 403:
```

Comment:
If it's not a permission denied, we should rethrow the exception, which will be handled a layer above